### PR TITLE
Phase-1: Migrate Netsuite to platform ( Expenses API )

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/fyle/platform_connector.py [hunk 1]
```diff
@@ -1,76 +1,40 @@
-import json
```
Comment (NileshPant1999) [inline apps/fyle/platform_connector.py:1]: can remove this entire file once all the tax-related changes in other SDKs are released.

File: apps/fyle/helpers.py [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 from apps.fyle.models import ExpenseGroupSettings
 from apps.mappings.models import GeneralMapping
-from apps.workspaces.models import Workspace
+from apps.workspaces.models import FyleCredential, Workspace
+from fyle_integrations_platform_connector import PlatformConnector
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply apps/fyle/helpers.py:9]: not a blocker comment, just a guideline

File: apps/fyle/models.py [hunk 3]
```diff
@@ -87,73 +99,58 @@ def create_expense_objects(expenses: List[Dict], workspace_id: int):
...
                         'sub_category': expense['sub_category'],
-                        'project': expense['project_name'],
+                        'project': expense['project'],
                         'project_id': expense['project_id'],
                         'expense_number': expense['expense_number'],
+                        'org_id': expense['org_id'],
+                        'tax_amount': expense['tax_amount'],
+                        'tax_group_id': expense['tax_group_id'],
                         'claim_number': expense['claim_number'],
-                        'amount': round(expense['amount'], 2),
```
Comment (ashwin1111) [inline apps/fyle/models.py:124]: why is the existing behaviour changed?

File: apps/fyle/tasks.py [hunk 4]
```diff
@@ -5,16 +5,22 @@
...
 
-from .models import Expense, ExpenseGroup, ExpenseGroupSettings
-from .connector import FyleConnector
+from .models import SOURCE_ACCOUNT_MAP, Expense, ExpenseGroup, ExpenseGroupSettings
+from .serializers import ExpenseGroupSerializer
 
 logger = logging.getLogger(__name__)
 logger.level = logging.INFO
 
+SOURCE_ACCOUNT_MAP = {
```
Comment (ashwin1111) [inline apps/fyle/tasks.py:20]: I see `SOURCE_ACCOUNT_MAP` is imported, why declare again

File: requirements.txt [hunk 5]
```diff
@@ -44,7 +44,6 @@ pytest-django==4.4.0
 pytest-mock==3.6.1
 python-dateutil==2.8.1
 pytz==2019.3
-requests==2.25.0
```
Comment (ashwin1111) [inline requirements.txt:47]: why is this removed?

File: tests/conftest.py [hunk 6]
```diff
@@ -71,5 +71,6 @@ def add_fyle_credentials(db):
     for workspace_id in workspaces:
         FyleCredential.objects.create(
             refresh_token=settings.FYLE_REFRESH_TOKEN,
-            workspace_id=workspace_id
+            workspace_id=workspace_id,
+            cluster_domain='http://staging.fyle.tech/'
```
Comment (ashwin1111) [inline tests/conftest.py:75]: https

File: tests/test_fyle/fixtures.py [hunk 7]
```diff
@@ -9,38 +9,41 @@
...
             "sub_category": "Food",
             "fyle_category": "Food",
             "settlement_id": "sethtvWoRZfHU",
             "expense_number": "E/2021/11/T/3",
             "claim_number": "C/2021/11/R/1",
             "external_id": None,
-            "tax": None,
+            "tax_amount": None,
             "tax_group_id": None,
             "trip_request_id": None,
```
Comment (ashwin1111) [inline tests/test_fyle/fixtures.py:33]: we don't get all these fields now

Comment (ashwin1111) [reply tests/test_fyle/fixtures.py:33]: I assume it doesn't affect any flow, still add a todo and cleanup later if it takes time

File: tests/test_fyle/fixtures.py [hunk 8]
```diff
@@ -525,7 +544,7 @@
         "description": {
             "report_id": "rpuN3bgphxbK",
             "fund_source": "PERSONAL",
-            "claim_number": "C/2021/11/R/5",
+            "claim_number": "C/2021/11/R/5 - ",
```
Comment (ashwin1111) [inline tests/test_fyle/fixtures.py:547]: is this for failing a test intentionally?

File: tests/test_fyle/test_helpers.py [hunk 9]
```diff
@@ -44,7 +44,7 @@ def test_update_use_employee_attributes_flag():
     assert general_mapping.use_employee_location == False
 
 @pytest.mark.django_db
-def test_check_interval_and_sync_dimension(test_connection):
+def test_check_interval_and_sync_dimension(test_connection, add_fyle_credentials):
```
Comment (ashwin1111) [inline tests/test_fyle/test_helpers.py:47]: add a todo, if this will be utilised later

File: tests/test_fyle/test_platform_connector.py [hunk 10]
```diff
@@ -1,36 +1,21 @@
 from fyle_accounting_mappings.models import ExpenseAttribute
 import pytest
 from apps.workspaces.models import FyleCredential, Workspace
-from apps.fyle.platform_connector import FylePlatformConnector
+from fyle_integrations_platform_connector import PlatformConnector
```
Comment (ashwin1111) [inline tests/test_fyle/test_platform_connector.py:4]: imports order

File: tests/test_fyle/fixtures.py (hunk 7 above)
Comment (NileshPant1999) [reply tests/test_fyle/fixtures.py:33]: yes @ashwin1111 it is a temporary fix will fix them later

File: apps/fyle/models.py [hunk 11]
```diff
@@ -87,73 +99,58 @@ def create_expense_objects(expenses: List[Dict], workspace_id: int):
...
             cutoff_date = _format_date('2021-06-15T00:00:00.000Z')
-            expense_settled_at = _format_date(expense['settled_at'])
-            if expense_settled_at > cutoff_date:
-                expense_custom_properties = {}
-
-                if custom_property_keys and expense['custom_properties']:
-                    for prop in expense['custom_properties']:
-                        if prop['name'].lower() in custom_property_keys:
-                            expense_custom_properties[prop['name']] = prop['value']
+            expense_created_at = _format_date(expense['expense_created_at'])
```
Comment (ashwin1111) [inline apps/fyle/models.py:108]: still this is not in alignment with the old behaviour

Comment (NileshPant1999) [reply apps/fyle/models.py:108]: @ashwin1111 i didn't get what you are trying to say here

File: apps/fyle/tasks.py [hunk 12]
```diff
@@ -90,7 +82,7 @@ def create_expense_groups(workspace_id: int, fund_source: List[str], task_log: T
             )
 
             task_log.status = 'COMPLETE'
-            task_log.detail = None
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply apps/fyle/tasks.py:93]: why we are saving it as None it is anyway going to be even without saving it right ??

File: apps/fyle/models.py (hunk 11 above)
Comment (ashwin1111) [reply apps/fyle/models.py:108]: These both are different right?
```
            expense_settled_at = _format_date(expense['settled_at'])
              if expense_settled_at > cutoff_date:
            
            expense_created_at = _format_date(expense['expense_created_at'])
              if expense_created_at > cutoff_date:

File: apps/fyle/tasks.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (ashwin1111) [reply apps/fyle/tasks.py:93]: Assume a call failed to fyle api, we store the exception response in the detail column.
Now, when the next run comes, assume it went through successfully. Now, we have those faulty values in detail column even though they went through successfully.

[the remark this answers is not in the document]
Comment (NileshPant1999) [reply apps/fyle/tasks.py:93]: ohh okay got it thanks 🙇🏻

File: apps/fyle/models.py (hunk 11 above)
Comment (NileshPant1999) [reply apps/fyle/models.py:108]: ``` currently not getting  settled_at via the platform connector```

File: tests/test_fyle/fixtures.py (hunk 7 above)
Comment (NileshPant1999) [reply tests/test_fyle/fixtures.py:33]: already fixed in next pr

**Review by ashwin1111 — APPROVED** [review]

File: apps/fyle/tasks.py [hunk 13]
```diff
@@ -50,37 +55,23 @@ def create_expense_groups(workspace_id: int, fund_source: List[str], task_log: T
...
 
-            last_synced_at = datetime.now()
-            expenses = fyle_connector.get_expenses(
-                state=import_state,
-                fund_source=fund_source,
-                settled_at=filter_by_timestamp if expense_group_settings.expense_state == 'PAYMENT_PROCESSING' \
-                    else None,
-                updated_at=filter_by_timestamp if expense_group_settings.expense_state == 'PAID' else None
+            expenses = platform.expenses.get(
+                source_account_type, expense_group_settings.expense_state, last_synced_at, False
```
Comment (ashwin1111) [inline apps/fyle/tasks.py:70]: this has to be fixed, last param

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: apps/fyle/tasks.py (hunk 13 above)
Comment (NileshPant1999) [reply apps/fyle/tasks.py:70]: okay

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**